### ISSUE TITLE
Multiline Matrix messages not correctly sent to the IRC channel

### ISSUE THREAD

**Issue by babolivier** [matrix-org/matrix-appservice-irc#488]
I was chatting in an IRC query (1:1 chat) with a friend, and sent him a 4-line message. What my friend received was:

> babolivier_m sent a long message: babolivier_m_2017-08-11_17:16:39.txt \<https://matrix.org/_matrix/media/v1/download/matrix.org/[redacted]>

I think that's not really nice in term or user experience (confirmed by my friend complaining about that). Having spent years on IRC, I'm used to IRC clients, when pasting a multiline message into them, splitting each line into a message, and that's the behavior I was expecting the bridge to have.

I also don't think it's a good idea to force an user to reach an external URL to read a part of the conversation.

Version: The current version deployed on matrix.org
IRC network: chat.freenode.net

**Comment by silkeh** [comment]
This behaviour is configurable with the [`lineLimit`](https://github.com/matrix-org/matrix-appservice-irc/blob/master/config.sample.yaml#L298) setting. For any number of lines below this limit the message is split up. The problem is that Matrix allows for very long messages and most IRC networks impose limits on the number of lines that can be sent within a certain amount of time.
There are also messages (eg: posting source code) for which it does make sense to link to the content.

**Comment by babolivier** [comment]
> The problem is that Matrix allows for very long messages and most IRC networks impose limits on the number of lines that can be sent within a certain amount of time.

Right, I didn't think about this limit. I understand the necessity for the `lineLimit` (which I didn't know about (I use matrix.org's bridge, didn't look into the config), thanks for that), but 3 (the default) seems really low to me. If I have time, I might do some research on the threshold for most popular IRC networks and PR a new default value.

> There are also messages (eg: posting source code) for which it does make sense to link to the content.

Right again. We could check the message if it includes source code (most of the time contained within markdown's "```") though. Again, I'll open a PR if I can find some time.

**Comment by Half-Shot** [comment]
This is going to be somewhat improved with #529
